toolchain/check/testdata/impl/fail_todo_impl_assoc_const.carbon

```diff
@@ -10,27 +10,51 @@
 
 interface I { let T:! type; }
 
-// CHECK:STDERR: fail_todo_impl_assoc_const.carbon:[[@LINE+3]]:1: error: semantics TODO: `impl of interface with associated constant`
+// CHECK:STDERR: fail_todo_impl_assoc_const.carbon:[[@LINE+10]]:1: error: semantics TODO: `impl of interface with associated constant`
 // CHECK:STDERR: impl bool as I where .T = bool {}
 // CHECK:STDERR: ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
+// CHECK:STDERR: fail_todo_impl_assoc_const.carbon:[[@LINE+6]]:27: error: cannot implicitly convert from `type` to `<associated type in I>`
```

Comment:
Would it be straightforward to have the TODO diagnostic above suppress these diagnostics, e.g. by checking for the `BuiltinError` witness that's produced when the TODO diagnostic is emitted?

Reply:
I don't think so, since I think they are actually generated in the opposite order.